From d835ba06e865097a6315be6180e8624ac0e3fb6a Mon Sep 17 00:00:00 2001
From: Wang Han <416810799@qq.com>
Date: Wed, 14 Nov 2018 15:48:09 +0800
Subject: [PATCH 205/211] Snap: Avoid crash with empty RAW output size

 * Check for length of rawSize, fixes front camera on oneplus2

Change-Id: I3640fde2e6649c715c26cf366d2b71679b59a83c
---
 src/com/android/camera/CaptureModule.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/camera/CaptureModule.java b/src/com/android/camera/CaptureModule.java
index 65e32c736..379446544 100755
--- a/src/com/android/camera/CaptureModule.java
+++ b/src/com/android/camera/CaptureModule.java
@@ -3777,7 +3777,7 @@ public class CaptureModule implements CameraModule, PhotoController,
         mSupportedMaxPictureSize = prevSizes[0];
         Size[] rawSize = mSettingsManager.getSupportedOutputSize(getMainCameraId(),
                     ImageFormat.RAW10);
-        if (rawSize != null) {
+        if (rawSize != null && rawSize.length > 0) {
             mSupportedRawPictureSize = rawSize[0];
         }
         mPreviewSize = getOptimalPreviewSize(mPictureSize, prevSizes);
-- 
2.17.1

